File: Aug/Day1/pascals-triangle.cc
//  Time: O(N^2)  ::  Space: O(N^2)
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascalList;
        vector<int> nums = {1};
        pascalList.push_back(nums);

        for (int i = 1; i < numRows; ++i) {
            vector<int> temp(nums.size() + 1, 1);
            for (int j = 1; j < temp.size() - 1; ++j) {
                temp[j] = nums[j] + nums[j - 1];
            }
            nums = temp;
            pascalList.push_back(nums);
        }
        return pascalList;
    }
};

File: Aug/Day1/pascals-triangle.java
//  Time: O(N^2)  ::  Space: O(N^2)
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascalList = new ArrayList<>();
        List<Integer> nums = new ArrayList<>();
        nums.add(1);
        pascalList.add(new ArrayList<>(nums));

        for (int i = 1; i < numRows; i++) {
            List<Integer> temp = new ArrayList<>();
            temp.add(1);
            for (int j = 1; j < nums.size(); j++) {
                temp.add(nums.get(j) + nums.get(j - 1));
            }
            temp.add(1);
            nums = temp;
            pascalList.add(new ArrayList<>(nums));
        }

        return pascalList;
    }
}
